### PR TITLE
Add script to automate Atlas Checks Spark job on EC2

### PR DESCRIPTION
**Pull request by atiannicelli**
### Description:

Add a script that automates the execution of Atlas Checks on an EC2 Instance. Please see README.md for details

### Potential Impact:

No impact to Atlas Check code.

### Unit Test Approach:

No Unit tests performed.

### Test Results:

This change was tested with our AWS setup that includes a template for Atlas Checks that we use to start EC2 instances. To complete this script I plan on adding a "Prep" command that will create an EC2 instance from scratch and install necessary support components to make this script work from a native EC2 instance. Until that is completed this script assumes that the user has a launch template with the correct environment to execute the Atlas Checks spark job.

### CODE REVIEW

File: scripts/cloud-check-control/cloudAtlasCheckControl.py [hunk 1]
```diff
@@ -0,0 +1,663 @@
...
+            # if configuration.json is a file then copy it to the EC2 instance
+            if self.atlasConfig.find("http") < 0:
+                atlasConfig = "file://" + os.path.join(self.homeDir, self.atlasConfig)
+
+            # sync local input folder with S3 input bucket and directory
+            # cmd = "aws s3 sync s3://{} {}".format(self.s3InFolder, self.atlasInDir)
+            # if self.ssh_cmd(cmd):
+            #     finish("Unable to pull atlas input files from S3", -1)
+
+            # TODO Call command to submit job
```
Comment (zlavergne) [inline scripts/cloud-check-control/cloudAtlasCheckControl.py:150]: Are these comments and TODO still useful? Consider deleting if not.

File: scripts/cloud-check-control/README.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+- `-i IN, --input IN` - The S3 Input directory where atlas files are located to execute checks on. This S3 bucket will be mounted to the EC2 instance for readonly processing of the atlas files.
+- `-p PROCESSES, --processes PROCESSES` - The number of parallel osmium processes to start. Note that when processing a large PBF file each osmium process will use a great deal of memory so small numbers of parallel processes is suggested. (Default: 32)
+- `-j CONFIG, --config CONFIG` - The json file to use as an [Atlas configuration file](https://github.com/osmlab/atlas-checks/blob/dev/docs/configuration.md) to control the Atlas Checks process. This can either be a fully qualified URL or a full path to a json file in the S3 Input bucket. (e.g my-s3-bucket/Atlas_Checks/configurations/special_config.json) (Default: https://raw.githubusercontent.com/osmlab/atlas-checks/dev/config/configuration.json)
+- `-f FORMATS --formats FORMATS` - A comma separated list of formats to use to determine the output format for Atlas Checks. (Default: 'flags')
+- `-m MEMORY,--memory MEMORY` - The Maximum amount of memory for the Spark job to use. (Default: 256)
```
Comment (zlavergne) [inline scripts/cloud-check-control/README.md:69]: let's specify that this is GB ("###g")

File: scripts/cloud-check-control/cloudAtlasCheckControl.py [hunk 3]
```diff
@@ -0,0 +1,663 @@
...
+    if hasattr(args, "formats") and args.formats:
+        cloudctl.formats = args.formats
+    if hasattr(args, "memory") and args.memory:
+        cloudctl.memory = args.memory
+    if hasattr(args, "config") and args.config:
+        cloudctl.atlasConfig = args.config
+    if hasattr(args, "id") and args.id:
+        cloudctl.instanceId = args.id
+        cloudctl.get_instance_info()
+    if args.version:
```
Comment (zlavergne) [inline scripts/cloud-check-control/cloudAtlasCheckControl.py:647]: Same considerations about truthiness in PBF shard script

**Review by zlavergne — CHANGES_REQUESTED** [review]
Nothing stood out except for the question about conditional checks for truthiness. The change isn't explicitly necessary but I want to make sure it's addressed at least in comments.

File: scripts/cloud-check-control/cloudAtlasCheckControl.py (hunk 1 above)
Comment (atiannicelli) [reply scripts/cloud-check-control/cloudAtlasCheckControl.py:150]: Ooops. removed.

File: scripts/cloud-check-control/cloudAtlasCheckControl.py [hunk 4]
```diff
@@ -0,0 +1,656 @@
...
+
+    def __init__(
+        self,
+        timeoutMinutes=6000,
+        key="",
+        instanceId="",
+        processes=32,
+        memory=256,
+        formats="flags",
+        countries="BRA",
```
Comment (Bentleysb) [inline scripts/cloud-check-control/cloudAtlasCheckControl.py:61]: Does it really make sense for this to have a default? Perhaps this should just be an empty string?

**Review by Bentleysb — COMMENTED** [review]
I know very little about EC2, but the code looks clean. I just had one question about the country default.

File: scripts/cloud-check-control/cloudAtlasCheckControl.py (hunk 4 above)
Comment (atiannicelli) [reply scripts/cloud-check-control/cloudAtlasCheckControl.py:61]: Removed this default and made -c or --countries a required parameter.

**Review by seancoulter — APPROVED** [review]
This looks great and well fleshed out. Thanks!

**Review by Bentleysb — APPROVED** [review]

**Pull request merged**